src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private int size;
+    private Object[] elementData;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_SIZE];
+
+    }
+
+    public void checkCapacity() {
+        if (size == elementData.length) {
+            int newSize = (Integer) elementData.length * 3 / 2;
```

Comment:
Casting `elementData.length` to `Integer` is unnecessary and can lead to a `ClassCastException`. The expression should be `elementData.length * 3 / 2` without casting.